//Client Side

message Init {
required int32 trunkSize = 1;
required string filename = 3;
}

message Resume {
}

message Pause{
}

message Stop{
}

message RequestStrongChecksum{
repeated int32 trunkIdx=1;
}

message RequestTrunks{
repeated int32 trunkIdx=1;
}

//Sended by Server Side

message Trunk {
required uint32 idx=1;
required bytes data = 2;
}

message WeakChecksum{
repeated uint32 sum=1;
required uint32 lastTrunkSize = 2; // 0 means without remainder
};

message StrongChecksum{
repeated bytes sum=1; //128-bit?
};
